genai-perf/genai_perf/TCO_calculator/TCO_calculator.ipynb

```diff
@@ -0,0 +1,621 @@
+    "            -m $MODEL \\\n",
+    "            --endpoint-type chat \\\n",
+    "            --service-kind openai \\\n",
+    "            --streaming \\\n",
+    "            -u localhost:8000 \\\n",
+    "            --synthetic-input-tokens-mean $INPUT_SEQUENCE_LENGTH \\\n",
+    "            --synthetic-input-tokens-stddev $INPUT_SEQUENCE_STD \\\n",
+    "            --concurrency $CONCURRENCY \\\n",
+    "            --output-tokens-mean $OUTPUT_SEQUENCE_LENGTH \\\n",
+    "            --extra-inputs max_tokens:$OUTPUT_SEQUENCE_LENGTH \\\n",
+    "            --extra-inputs min_tokens:$OUTPUT_SEQUENCE_LENGTH \\\n",
+    "            --extra-inputs ignore_eos:true \\\n",
+    "            --tokenizer meta-llama/Meta-Llama-3-8B-Instruct \\\n",
+    "            --measurement-interval 30000 \\\n",
+    "            --profile-export-file ${INPUT_SEQUENCE_LENGTH}_${OUTPUT_SEQUENCE_LENGTH}.json \\\n",
+    "            -- \\\n",
+    "            -v \\\n",
+    "            --max-threads=256\n",
+    "    \n",
+    "    done\n",
+    "}\n",
+    "\n",
+    "# Iterate over all defined use cases and run the benchmark script for each\n",
+    "for description in \"${!useCases[@]}\"; do\n",
+    "    runBenchmark \"$description\"\n",
+    "done\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "603f1941-5206-4bca-a547-028e0ea50f21",
+   "metadata": {},
+   "source": [
+    "Next, we execute the bash script, which will carry out the defined benchmarking scenarios and gather the data in a default directory named `artifacts` under the current working directory."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6cbfacd3-5755-4c0b-ae23-3abffceebbdb",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "bash benchmark.sh"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c480b28d-6816-4c84-9124-bdc56fc81f41",
+   "metadata": {},
+   "source": [
+    "## Reading gen-AI-perf data\n",
+    "\n",
+    "Once performance benchmarking is done, we read and collect the results in a single data frame."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "3815b7f1-b51b-44ea-bbbc-b023ec0beeca",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "gen_AI_perf_field = [\n",
+    " 'Inter Token 90th Percentile Latency (ms)',\n",
+    " 'Inter Token 99th Percentile Latency (ms)',\n",
+    " 'Inter Token Average Latency (ms)',\n",
+    " 'Time to First Token 90th Percentile Latency (ms)',\n",
+    " 'Time to First Token 99th Percentile Latency (ms)',\n",
+    " 'Time to First Token Average Latency (ms)',\n",
+    " 'Request 90th Percentile Latency (ms)',\n",
+    " 'Request 99th Percentile Latency (ms)',\n",
+    " 'Request Latency (ms)',\n",
+    " 'Requests per Second',\n",
+    " 'Tokens per Second']\n",
+    "\n",
+    "# Other experimental params: 'Seq Length (ISL/OSL)', 'Concurrency',"
```

Comment:
Why are these listed separate and not as part of the above definition? The fields appear to be used by the code below.

Reply:
these are not part of GenAI-perf data. It's managed outside as defined by the bash script

Reply:
restructured the code and removed this cell